src/Microsoft.ML.LightGbm/LightGbmTrainerBase.cs

```diff
@@ -436,7 +445,7 @@ private protected virtual void GetDefaultParameters(IChannel ch, int numRow, boo
 
         private FloatLabelCursor.Factory CreateCursorFactory(RoleMappedData data)
         {
-            var loadFlags = CursOpt.AllLabels | CursOpt.Features;
+            var loadFlags = CursOpt.AllLabels | CursOpt.AllFeatures;
```

Comment:
>AllFeatures [](start = 56, length = 11)

What does LightGBM do if `HandleMissingValues` is false? If it ignores examples with missing values, perhaps we should filter them on our end in that case?

Reply:
I believe it does ignore examples with missing values when HandleMissingValues is false. When it is true, the missing values are imputed.

Reply:
If LightGBM is going to ignore these examples anyway, we might as well not pass them to LightGBM in the first place.

---
In reply to: [371399038](https://github.com/dotnet/machinelearning/pull/4695#discussion_r371399038) [](ancestors = 371399038)

Reply:
LightGBM's missing value handling won't cause the entire row to be ignored. 

When not specifically handling a missing value, (I believe) the one missing value in the row is not counted in the histogram for the feature. The non-missing values in the row are still be counted in the histograms for their respective features.